### PR TITLE
[VCDA-2956]Fix DNAT rule not being deleted during rollback

### PR DESCRIPTION
**Pull request by Anirudh9794**
Signed-off-by: Aniruddha Shamasundar <aniruddha.9794@gmail.com>

To help us process your pull request efficiently, please include: 

Split add_control_plane function as exposed IP won't be propagated to _create_cluster_async() if there is an error during cluster creation.

Testing done:
* Tested DNAT rule deletion by simulating an error during cluster creation
* Tested if TKGm cluster is created successfully if there is no error

@rocknes @sakthisunda

---
This change is [Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/1218)

### CODE REVIEW

**Review by sakthisunda — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/1218)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @Anirudh9794 and @sakthisunda)

___
*[container_service_extension/rde/backend/cluster_service_2_x_tkgm.py, line 2066 at r1](https://reviewable.io/reviews/vmware/container-service-extension/1218#-MkszNpP0aDb36Nff-pi:-MkszNpP0aDb36Nff-pj:bunvkvr) ([raw file](https://github.com/vmware/container-service-extension/blob/6d82e60bbb39b2e75ed9b3f43da994bc97bfa183/container_service_extension/rde/backend/cluster_service_2_x_tkgm.py#L2066)):*
> ```Python
> 
>         # Get template with no expose_ip; expose_ip will be computed
>         # later when control_plane internal ip is computed below.
> ```

The above two line comments need to be removed or moved to the logic that handles internal vs external ip in <b>_create_cluster_async()</b>
___
*[container_service_extension/rde/backend/cluster_service_2_x_tkgm.py, line 2085 at r1](https://reviewable.io/reviews/vmware/container-service-extension/1218#-Mkt-I562ImvWVRqGmAj:-Mkt-I575G1JEoHzQIY0:b44tpx) ([raw file](https://github.com/vmware/container-service-extension/blob/6d82e60bbb39b2e75ed9b3f43da994bc97bfa183/container_service_extension/rde/backend/cluster_service_2_x_tkgm.py#L2085)):*
> ```Python
>                 vcd_host=vcd_host.replace("/", r"\/"),
>                 org=org.get_name(),
>                 vdc=vdc.name,
> ```

👍

**Review by sakthisunda — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/1218)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @Anirudh9794 and @sakthisunda)

*[a discussion](https://reviewable.io/reviews/vmware/container-service-extension/1218#-Mkt0z9L2IFqY_PU5tDK:-Mkt0z9M9vGxCmvfQ90F:bjhyft8) (no related file):*
expose_ip needs to be having a right value in the rollback logic. This code change sets the right value of expose_ip for rollback to work correctly trigger DNAT rule removal. thanks

---

**Review by Anirudh9794 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/container-service-extension/1218)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @Anirudh9794 and @sakthisunda)

___
*[container_service_extension/rde/backend/cluster_service_2_x_tkgm.py, line 2066 at r1](https://reviewable.io/reviews/vmware/container-service-extension/1218#-MkszNpP0aDb36Nff-pi:-Mkt4sL67gRvXoqPdOEv:b-896fix) ([raw file](https://github.com/vmware/container-service-extension/blob/6d82e60bbb39b2e75ed9b3f43da994bc97bfa183/container_service_extension/rde/backend/cluster_service_2_x_tkgm.py#L2066)):*
<details><summary><i>Previously, sakthisunda wrote…</i></summary><blockquote>

The above two line comments need to be removed or moved to the logic that handles internal vs external ip in <b>_create_cluster_async()</b>
</blockquote></details>

Done.
___
*[container_service_extension/rde/backend/cluster_service_2_x_tkgm.py, line 2085 at r1](https://reviewable.io/reviews/vmware/container-service-extension/1218#-Mkt-I562ImvWVRqGmAj:-Mkt1Uk70QSy-MHLvvj7:b-896fix) ([raw file](https://github.com/vmware/container-service-extension/blob/6d82e60bbb39b2e75ed9b3f43da994bc97bfa183/container_service_extension/rde/backend/cluster_service_2_x_tkgm.py#L2085)):*
<details><summary><i>Previously, sakthisunda wrote…</i></summary><blockquote>

👍
</blockquote></details>

Done.

File: container_service_extension/rde/backend/cluster_service_2_x_tkgm.py [hunk 1]
```diff
@@ -2033,8 +2036,8 @@ def _add_control_plane_nodes(
         for spec in vm_specs:
             spec['cust_script'] = templated_script.format(
                 vcd_host=vcd_host.replace("/", r"\/"),
-                org=org,
-                vdc=vdc,
+                org=org.get_name(),
```
Comment (arunmk) [inline container_service_extension/rde/backend/cluster_service_2_x_tkgm.py:2039]: The changes in these two lines must be undone, right?

**Review by arunmk — APPROVED** [review]
I think two changes need to be undone. The rest is fine.

File: container_service_extension/rde/backend/cluster_service_2_x_tkgm.py (hunk 1 above)
Comment (Anirudh9794) [reply container_service_extension/rde/backend/cluster_service_2_x_tkgm.py:2039]: As discussed, org.get_name() and vdc.name are the correct parameters

**Pull request merged**